map_model/src/pathfind/pathfinder.rs

```diff
@@ -40,82 +38,30 @@ impl Pathfinder {
-                        .allow_through_traffic
-                        .contains(req.constraints)
-                    {
-                        if req.constraints == PathConstraints::Pedestrian {
-                            let steps =
-                                walking_path_to_steps(z1.pathfind_walking(req.clone(), map)?, map);
-                            return Some(Path::new(map, steps, req, Vec::new()));
-                        }
-                        return z1.pathfind(req, map);
-                    }
-                } else {
-                    // TODO Handle paths going between two different zones
-                    return None;
-                }
-            }
-            (Some(zone), None) => {
-                if !zone
-                    .restrictions
-                    .allow_through_traffic
-                    .contains(req.constraints)
-                {
-                    // Calculate the entire path using every possible border, then take the one
-                    // with the least total distance.
-                    // TODO This is slow and doesn't account for the mode-specific cost.
-                    let mut paths = Vec::new();
-                    for i in &zone.borders {
-                        if let Some(result) =
-                            self.pathfind_from_zone(map.get_i(*i), req.clone(), zone, map)
-                        {
-                            paths.push(result);
-                        }
-                    }
-                    return paths.into_iter().min_by_key(|p| p.total_length());
-                }
-            }
-            (None, Some(zone)) => {
-                if !zone
-                    .restrictions
-                    .allow_through_traffic
-                    .contains(req.constraints)
-                {
-                    // Calculate the entire path using every possible border, then take the one
-                    // with the least total distance.
-                    // TODO This is slow and doesn't account for the mode-specific cost.
-                    let mut paths = Vec::new();
-                    for i in &zone.borders {
-                        if let Some(result) =
-                            self.pathfind_to_zone(map.get_i(*i), req.clone(), zone, map)
-                        {
-                            paths.push(result);
-                        }
-                    }
-                    return paths.into_iter().min_by_key(|p| p.total_length());
-                }
-            }
-            (None, None) => {}
-        }
-
         if req.constraints == PathConstraints::Pedestrian {
             if req.start.lane() == req.end.lane() {
                 return Some(one_step_walking_path(&req, map));
             }
-            let steps = walking_path_to_steps(self.simple_walking_path(&req, map)?, map);
+            let nodes = match self {
+                Pathfinder::Dijkstra => dijkstra::simple_walking_path(&req, map)?,
```

Comment:
Can you elaborate on when we expect to be using Dijkstra at this point?

Reply:
Only if you import a map passing `--skip_ch`. None of the "official" maps in S3 use this; I just do it to debug occasionally. I could add a comment here, but the docstring on `Pathfinder` above sort of answers this.

... One more slight exception, `pathfind_avoiding_lanes` uses Dijkstra always. This gets called when congestion caps (an experimental feature) are applied through map edits and/or flags to the sim. I'll add a comment there.

Reply:
gotcha, thanks!